htsget/htsget.go

```diff
@@ -0,0 +1,238 @@
+package htsget
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/NBISweden/sda-cli/helpers"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help htsget` command
+var Usage = `
+USAGE: %s htsget [-dataset <datasetID>] [-filename <filename>] (-reference <referenceName>) [-htsgethost <htsget-hostname>] [-key <public-key-file>] (-outdir <dir>)
+
+htsget:
+    Htsget downloads files from the Sensitive Data Archive (SDA), using the
+	htsget server. A dataset and a filename must be provided in order to 
+	download the file. The files will be downloaded in the current
+    directory, if outdir is not defined.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help
+var ArgHelp = `
+    [dataset]
+        The ID of the dataset that the file is part of.
+    [filename]
+        The name of the file to download.
+	[reference]
+		The reference number of the file to download.
+	[htsgethost]
+		The hostname of the htsget server to use.
+	[key]
+		The public key file to use for the htsget request.`
+
+// Args is a flagset that needs to be exported so that it can be written to the
+// main program help
+var Args = flag.NewFlagSet("htsget", flag.ExitOnError)
+var datasetID = Args.String("dataset", "", "Dataset ID for the file to download")
+var fileName = Args.String("filename", "", "The name of the file to download")
+var referenceName = Args.String("reference", "", "The reference number of the file to download")
```

Comment:
Can the user also specify the range of coordinates of the BAM file?
Maybe with the format `referenceName:beg:end` or use another argument `-range`.

Reply:
That sounds reasonable, I'll check if we can do it.

Reply:
I tested this format `referenceName:beg:end` a bit and it produces the same results with the case of not providing any reference at all.
I will skip adding this now, instead we can address it on the follow up work for the vcf support.